jelares/portfolio/src/main/java/com/google/sps/servlets/BlogDataServlet.java

```diff
@@ -0,0 +1,159 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.servletData.BlogPost;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that manages the blog page */
+@WebServlet("/blog-data")
+public class BlogDataServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Read the query string to get post limit
+    int postLimit = Integer.parseInt(request.getParameter("num-posts"));
+
+    // Query to find all post entities sorted from newest to oldest
+    Query query = new Query("blog-post").addSort("timestamp", SortDirection.ASCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<BlogPost> blogPosts = new ArrayList<>();
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(postLimit))) {
+      String imageUrl = (String) entity.getProperty("blog-post-image");
+      String header = (String) entity.getProperty("blog-post-title");
+      String content = (String) entity.getProperty("blog-post-content");
+
+      blogPosts.add(new BlogPost(imageUrl, header, content));
+    }
+
+    response.setContentType("application/json;");
+    String jsonBlog = blogToJson(blogPosts);
```

Comment:
Nit: `jsonBlog` -> `jsonBlogPost` would be more specific (since "blog" could mean multiple posts).

Reply:
In this case, jsonBlog does mean multiple posts encoded as a Json object

Reply:
That is a great point - I changed the name to what your suggested to reflect this.